### PR TITLE
Add `content-tree` extension

### PR DESCRIPTION
**Pull request by azer**

### CODE REVIEW

File: samples/content-tree/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+        fields[key].onValueChanged(this.onViewingEntryUpdated)
+      )
+    }
+
+    // Listen sys changes
+    this.detachFns.push(
+      this.props.extension.entry.onSysChanged(this.onViewingEntryUpdated)
+    )
+
+    await this.setContentTypeList()
```
[the remark this answers is not in the document]
Comment (azer) [reply samples/content-tree/src/index.jsx]: Yup, they are sequential . I think I'd prefer the simplicity of  `await` per line  over `Promise.all` abstraction in this case.

File: samples/content-tree/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,389 @@
+import { default as debounce } from "debounce-fn"
```
[the remark this answers is not in the document]
Comment (azer) [reply samples/content-tree/src/index.jsx]: Yes, fixed

**Comment by azer** [comment]
Comments addressed

File: samples/content-tree/build.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const fs = require('fs');
```
Comment (jelz) [inline samples/content-tree/build.js]: This file is not needed

File: samples/content-tree/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,12869 @@
+{
+  "name": "content-tree",
```
Comment (suevalov) [inline samples/content-tree/package-lock.json]: Let's remove package-lock.json from the repository

**Comment by suevalov** [comment]
@azer @jelz All dependencies have locked versions so `package-lock.json` is not needed

**Pull request merged**